src/Application/GamePathResolver.cpp

```diff
@@ -0,0 +1,79 @@
+﻿#include <vector>
+
+#include "Application/GamePathResolver.h"
+
+#include "Engine/IocContainer.h"
+#include "Library/Logger/Logger.h"
+
+
+using namespace Application;
+using EngineIoc = Engine_::IocContainer;
+
+
+static std::string _ResovlePath(Platform *platform, const char *envVarOverride, const std::vector<const wchar_t*> &registryKeys);
+
+
+std::string Application::ResolveMm6Path(Platform * platform) {
+    return _ResovlePath(
+        platform,
+        mm6PathOverrideKey,
+        {
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/Games/1207661253/PATH",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/GOGMM6/PATH",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/New World Computing/Might and Magic® VI/1.0/AppPath",
```

Comment:
It would be better if you hex-escaped the ® here. Use \x00AE.

Reply:
fair enough